autotest/ogr/data/csv/unbalanced_double_quotes.csv

```diff
@@ -0,0 +1,3 @@
+id,txt
+1,"foo""
+2,bar
```

Comment:
How do you define unbalanced?
`1,"foo""""` with 5 `"` is balenced and means string `foo"`?
Is it allowed that strings contain newlines?
```
1,"foo
2,bar"
```

Reply:
> How do you define unbalanced?

Rule: all starting double quote should have a matching ending double quote, and repeated double-quote inside a string is an escaped double uote.
Unbalanced: when the above is not true in a file

Reply:
> Is it allowed that strings contain newlines?
> 
> ```
> 1,"foo
> 2,bar"
> ```

yes, at least the OGR CSV driver allows it (which explains quite a bunch of complication in the code)